src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,86 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+
+            // Yield the paths of the StorageItems
+            foreach (IStorageItem item in storageFiles)
+            {
+                yield return item.Path;
+            }
         }
 
         public override IEnumerable<FileSystemInfo> EnumerateFileSystemInfos(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new FileSystemInfo[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+
+            // Translate the StorageFile results into FileSystemInfos
+            foreach (IStorageItem item in storageFiles)
+            {
+                if (item.IsOfType(StorageItemTypes.Folder))
+                {
+                    yield return new DirectoryInfo(item.Path);
+                }
+                else // If it is neither a File nor folder then we treat it as a File.
```

Comment:
Yes it is accurate. It's saying that if a StorageItem isn't a folder then we treat it as a file, regardless of whether it is actually a file.